examples/diagnostic_script.py

```diff
@@ -0,0 +1,195 @@
+    extended_attributes_asset = []
+    for asset in assets:
+        for item in asset["attributes"]:
+            if not item.startswith("arc_"):
+                extended_attributes_asset.extend(asset)
+    return extended_attributes_asset
+
+
+def attachment_identities_assets(assets_with_attachments: list) -> list:
+    """Create list of attachment identities"""
+    total_attachments_assets = set()
+    for asset in assets_with_attachments:
+        for item in asset["attributes"]["arc_attachments"]:
+            total_attachments_assets.add(item["arc_attachment_identity"])
+    return total_attachments_assets
+
+
+def events_ext_attr(events: list) -> list:
+    """Create list of events with extended attribute(s)"""
+    extended_attributes_event = []
+    for event in events:
+        for item in event["event_attributes"]:
+            if not item.startswith("arc_"):
+                extended_attributes_event.extend(event)
+    return extended_attributes_event
+
+
+def attachment_identities_events(events_with_attachments: list) -> list:
+    """Create list of attachment identities"""
+    total_attachments_events = set()
+    for event in events_with_attachments:
+        for item in event["event_attributes"]["arc_attachments"]:
+            total_attachments_events.add(item["arc_attachment_identity"])
+    return total_attachments_events
+
+
+def main():
+    """
+    Summarize usage of RKVST.
+    """
+
+    parser = argparse.ArgumentParser(description="RKVST Diagnostic Tool")
+
+    parser.add_argument(
+        "--url", "-u", default="https://app.rkvst.io", help="RKVST URL to access"
+    )
+    parser.add_argument(
+        "--client_id",
+        "-c",
+        default=os.environ.get("RKVST_CLIENT_ID"),
+        help="Specify Application CLIENT_ID inline",
+    )
+    parser.add_argument(
+        "--client_secret",
+        "-s",
+        default=os.environ.get("RKVST_CLIENT_SECRET"),
+        help="Specify Application CLIENT_SECRET inline",
+    )
+
+    args = parser.parse_args()
+
+    arch = Archivist(
+        args.url,
+        (args.client_id, args.client_secret),
+    )
+
+    props = {"confirmation_status": "CONFIRMED"}
+    attrs = {}  # attributes can be added to filer by name, type, etc.
+
+    # Total Number of Assets
+    assets = list(arch.assets.list(props=props, attrs=attrs))
+
+    # Assets with Extended Attributes
+    extended_attributes_asset = assets_ext_attr(assets)
+
+    # Assets with Associated Location
+    assets_with_location = list(
+        arch.assets.list(
+            props=props,
+            attrs=dict(
+                list(attrs.items()) + list({"arc_home_location_identity": "*"}.items())
+            ),
+        )
+    )
+
+    # Assets with Attachments
+    assets_with_attachments = list(
+        arch.assets.list(
+            props=props,
+            attrs=dict(list(attrs.items()) + list({"arc_attachments": "*"}.items())),
+        )
+    )
```

Comment:
Changed to use of _deepmerge and list comp for assets and events with attachments. I kept the merge with 'attrs' because one of the original requirements of the script was for users to have the option to filter their estate using attrs/props before running script.